x-pack/plugins/apm/server/lib/service_map/get_service_map_from_trace_ids.ts

```diff
@@ -29,21 +25,19 @@ export function getConnections(
   if (serviceName || environment) {
     paths = paths.filter((path) => {
       return path.some((node) => {
-        let matches = true;
-        if (serviceName) {
-          matches =
-            matches &&
-            SERVICE_NAME in node &&
-            (node as ServiceConnectionNode)[SERVICE_NAME] === serviceName;
+        if (serviceName && node[SERVICE_NAME] !== serviceName) {
+          return false;
         }
-        if (environment && environment !== ENVIRONMENT_NOT_DEFINED.value) {
-          matches =
-            matches &&
-            SERVICE_ENVIRONMENT in node &&
-            (node as ServiceConnectionNode)[SERVICE_ENVIRONMENT] ===
-              environment;
+
+        if (environment) {
+          if (environment === ENVIRONMENT_NOT_DEFINED.value) {
+            return !node[SERVICE_ENVIRONMENT];
+          }
+
+          return node[SERVICE_ENVIRONMENT] === environment;
         }
-        return matches;
+
+        return true;
```

Comment:
Much cleaner 👍 
This might be a small improvement (you decide)

```ts
return path.some((node) => {
  if (serviceName && node[SERVICE_NAME] !== serviceName) {
    return false;
  }

  if (!environment) {
    return true;
  }

  if (environment === ENVIRONMENT_NOT_DEFINED.value) {
    return !node[SERVICE_ENVIRONMENT];
  }

  return node[SERVICE_ENVIRONMENT] === environment;
});
```